pkg/osutil/osutil.go

```diff
@@ -149,11 +151,78 @@ func replaceTemplateVariables(fileSys fs.FS, srcPath string, draftConfig *config
 	}
 
 	fileString := string(file)
-
 	for _, variable := range draftConfig.Variables {
 		log.Debugf("replacing %s with %s", variable.Name, variable.Value)
 		fileString = strings.ReplaceAll(fileString, "{{"+variable.Name+"}}", variable.Value)
 	}
-
 	return []byte(fileString), nil
 }
+
+// CopyDirWithTemplates - Handles Gotemplate processing and writing
+func CopyDirWithTemplates(
+	fileSys fs.FS,
+	src, dest string,
+	variableMap map[string]string,
+	templateWriter templatewriter.TemplateWriter) error {
+
+	files, err := fs.ReadDir(fileSys, src)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+
+		if f.Name() == "draft.yaml" {
+			continue
+		}
+
+		srcPath := path.Join(src, f.Name())
+		destPath := path.Join(dest, f.Name())
+		log.Debugf("Source path: %s Dest path: %s", srcPath, destPath)
+
+		if f.IsDir() {
+			if err = templateWriter.EnsureDirectory(destPath); err != nil {
+				return err
+			}
+			if err = CopyDirWithTemplates(fileSys, srcPath, destPath, variableMap, templateWriter); err != nil {
+				return err
+			}
+		} else {
+			fileContent, err := replaceGoTemplateVariables(fileSys, srcPath, variableMap)
+			if err != nil {
+				return err
+			}
+
+			if err = checkAllVariablesSubstituted(string(fileContent)); err != nil {
```

Comment:
I'm not totally bought in that this will check all the variables are substituted from the go templates. Can you write some tests for this function that show it will work correctly for them?